sway-core/src/semantic_analysis/ast_node/expression/intrinsic_function.rs

```diff
@@ -166,30 +167,32 @@ fn type_check_encode_buffer_empty(
     Ok((kind, return_type))
 }
 
-fn encode_buffer_type(engines: &Engines) -> TypeInfo {
-    let raw_ptr = engines.te().insert(engines, TypeInfo::RawUntypedPtr, None);
+fn encode_buffer_type(engines: &Engines, span: Span) -> TypeInfo {
+    let raw_ptr = engines
+        .te()
+        .insert(engines, TypeInfo::RawUntypedPtr, span.source_id());
     let uint64 = engines.te().insert(
         engines,
         TypeInfo::UnsignedInteger(IntegerBits::SixtyFour),
-        None,
+        span.source_id(),
```

Comment:
Same here and on other places below. `u64`, `Unknown`, `Error` etc should have `None` as `source_id` . In all of the changes except the last one, I see `None` as the right input. Also, I see no point in specifying the same `span` for all the different type arguments because those types `raw_ptr` and `u64` do not have spans pointing at their declarations.

Reply:
I reverted this change.